Assignment 4

```diff
@@ -0,0 +1,26 @@
+Assignment 4 Sheqi Zhang
+
+1. Discuss the differences between strings and a character array.
+Answer:
+A string is an array of chars, and an array is a group of characters within the same type. An array is static, and we can't change its size after declarition, while we can change the strings after declaeition. There is always a "\0" at the end of a string, while an array doesn't have that. 
+
+2. Explain the advantages and disadvantages of arrays in C.
+Answer:
+Advantage: We can use it to store and use a group of similar data.
+Disadvantage: We can't change the property like the size of the array, which will make our programming not convenient.
+
+3. When does the compiler not implicitly generate the address of the first element of an array?
```

Comment:
So the short answer is that a compiler doesn't generate the address whenever it appears in an expression.